dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/converter/TrackerConverterService.java

```diff
@@ -46,4 +46,9 @@
     List<To> from( TrackerPreheat preheat, List<From> objects );
 
```

Comment:
if we had 

```java
    To from( TrackerPreheat preheat, From object, To existingInstance );

    default To from( TrackerPreheat trackerPreheat, Function<TrackerPreheat, To> toInstanceProvider, From object )
    {
        To instance = toInstanceProvider.apply( trackerPreheat );
        return from( trackerPreheat, object, instance );
    }

    default List<To> from( TrackerPreheat preheat, List<From> objects )
    {
        return objects
            .stream()
            .map( object -> from( preheat, object ) )
            .collect( Collectors.toList() );
    }

```
we might remove all implementations of these methods from children classes, whose responsibility would just be to implement:

```java
    To from( TrackerPreheat preheat, From object, To existingInstance );
```

Reply:
We didn't change the Converter Service directly. I agree with your comment, but I prefer to do it in a separate PR, since this one is already homungous.